clap_builder/src/builder/arg.rs

```diff
@@ -4363,8 +4363,20 @@ impl Arg {
         }
 
         debug_assert!(self.is_takes_value_set());
+
+        let is_required = |n| match self.is_positional() {
+            true => required && (num_vals.min_values() != 0),
+            false => {
+                // If all values are optional, the [] get rendered by the caller:
+                //    --foo[=<bar>]
+                // In this case, treat as required.
+                let required = self.get_min_vals() == 0;
+                required || (n < num_vals.min_values())
+            }
+        };
+
         for (n, val_name) in val_names.iter().enumerate() {
-            let arg_name = if self.is_positional() && (num_vals.min_values() == 0 || !required) {
+            let arg_name = if !is_required(n) {
```

Comment:
Could you clarifying the meaning, like `is_value_required` or `is_value_name_required`?  `required` on its own like this would make someone think of `Arg::required`.

Reply:
Sure, will do!

Reply:
Done, renamed to `value_is_required`.